internal/helper/structure.go

```diff
@@ -0,0 +1,34 @@
+package helper
+
+import "github.com/snapcore/snapd/gadget"
+
+// IsRootfsStructure determines if the given structure is the one associated
+// to the rootfs
+func IsRootfsStructure(s *gadget.VolumeStructure) bool {
+	if s == nil {
+		return false
+	}
+	return s.Role == gadget.SystemData
+}
+
+// IsSystemBootStructure determines if the given structure is the system-boot
+// one
+func IsSystemBootStructure(s *gadget.VolumeStructure) bool {
+	if s == nil {
+		return false
+	}
+
+	return s.Role == gadget.SystemBoot || s.Label == gadget.SystemBoot
+}
+
+// ShouldSkipStructure returns whether a structure should be skipped during certain processing
```

Comment:
See `handleSystemSeed` to understand what we mean by "seeded" here. This our way to track if the volume/image we work on contains a system-seed structure. This is the major "low-level" indicator throughout the code to adapt what we are doing (partition creation, define what we consider the "rootfs" content, etc.) and, in a way, to know if we are dealing with a UC image or a classic one. This is a bit ugly, and forces us to have too much `if/else` all around but a major refactoring of this is not on the table for now.
I was intentionally vague here because a precise list will be quickly outdated and I am no sure the documentation of this function is the place to do that.

Reply:
Hmm ok. Agree this is not the place to fix any of this.